pkgs/applications/editors/tmcbeans/default.nix

```
@@ -0,0 +1,70 @@
+{ lib, stdenv, fetchurl, makeWrapper, makeDesktopItem, which, unzip,
+libicns, imagemagick, jdk, perl
+}:
+
+let
+  version = "1.4.0";
+  desktopItem = makeDesktopItem {
+    name = "tmcbeans";
+    exec = "tmcbeans";
+    comment = "Integrated Development Environment";
+    desktopName = "TMCBeans IDE";
+    genericName = "Integrated Development Environment";
+    categories = [ "Development" ];
+    icon = "tmcbeans";
+  };
+in
+stdenv.mkDerivation {
+  pname = "tmcbeans";
+  inherit version;
+  src = fetchurl {
+    url = "https://download.mooc.fi/tmcbeans/installers/tmcbeans_${version}.zip";
+    hash = "sha512-kugZ7DAarbKK8RvOx7VYIpCwNhm3QVBlvqy+R7oeDDTT0LeDtHsWNIpIQSgT7t9MExfG1X9bqCqspJwYQxUG5g==";
+  };
+
+  buildCommand = ''
+    # Unpack and perform some path patching.
+    unzip $src
+    patchShebangs .
+
+    rm tmcbeans/bin/*.exe
```

There are more `exe` files.
```suggestion
    find -name \*.exe -exec rm -f {} \;
```